File: api/src/endpoint/question/vote_question.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::question::VoteResource};

pub struct VoteQuestion;

impl Endpoint for VoteQuestion {
    const PATH: &'static str = "/repository/{owner}/{repo}/question/{number}/vote";
    const METHOD: http::Method = http::Method::POST;

    type Request = VoteQuestionRequest;
    type Response = VoteQuestionResponse;
}

#[derive(Debug, Serialize, Deserialize)]
pub struct VoteQuestionRequest {
    pub value: i16,
}

pub type VoteQuestionResponse = VoteResource;

File: gitdot-api-ts/src/endpoint/question/vote-question.ts
import { z } from "zod";

import { VoteResource } from "../../resource";

export const VoteQuestionRequest = z.object({
  value: z.number().int(),
});
export type VoteQuestionRequest = z.infer<typeof VoteQuestionRequest>;

export const VoteQuestion = {
  path: "/repository/{owner}/{repo}/question/{number}/vote",
  method: "POST",
  request: VoteQuestionRequest,
  response: VoteResource,
} as const;
export type VoteQuestion = typeof VoteQuestion;
